### PR TITLE
[ESIMD] Provide alternative version of subscript operator

### PR DESCRIPTION
**Pull request by DenisBakhvalov**

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp [hunk 1]
```diff
@@ -336,6 +336,15 @@ template <typename BaseTy, typename RegionTy> class simd_view_impl {
     return v[i];
   }
 
+  /// Read a single element from a 1D region, by value only.
+  template <typename T = simd_view_impl,
+            typename = sycl::detail::enable_if_t<T::is1D()>>
+  __SYCL_DEPRECATED("use operator[] form.")
+  element_type operator()(int i) const {
```
Comment (kychendev) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp:343]: This only supports returning a scalar value with () operator. Is there a change to add support for returning a writable view from simd_view as requested?

Comment (DenisBakhvalov) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp:343]: @kychendev, yes, I'm working on that.

Comment (DenisBakhvalov) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp:343]: @kychendev, it turned out to be not that easy to do. I prefer to do it in further patches. But I'm working on it in the meantime.

**Review by kychendev — APPROVED** [review]

**Comment by DenisBakhvalov** [comment]
@kbobrovs is on vacation but we discussed this change before, he is aware of that.

**Comment by DenisBakhvalov** [comment]
@kbobrovs is on vacation but we discussed this change before, he is aware of that.
@bader, please merge the PR.

**Pull request merged**